gui/src/main/java/com/devonfw/ide/gui/IdeGuiContext.java

```diff
@@ -0,0 +1,34 @@
+package com.devonfw.ide.gui;
+
+import java.nio.file.Path;
+
+import com.devonfw.tools.ide.context.AbstractIdeContext;
+import com.devonfw.tools.ide.context.IdeStartContextImpl;
+import com.devonfw.tools.ide.io.IdeProgressBar;
+
+/**
+ *
+ */
+public class IdeGuiContext extends AbstractIdeContext {
+
+
+  /**
+   * The constructor.
+   *
+   * @param startContext the {@link com.devonfw.tools.ide.log.IdeLogger IdeLogger}.
+   * @param workingDirectory the optional {@link Path} to current working directory.
+   */
+  public IdeGuiContext(IdeStartContextImpl startContext, Path workingDirectory) {
+    super(startContext, workingDirectory);
+  }
+
+  @Override
+  protected String readLine() {
+    return "";
+  }
+
+  @Override
+  public IdeProgressBar newProgressBar(String title, long size, String unitName, long unitSize) {
+    return null;
```

Comment:
you could at least return a dummy implementation (see `IdeProgressBarTestImpl` that is however not available since it currently only exists in `test` code of `cli` module).

Reply:
I created issue #1296 for the entire topic, so we can close this now.